libursa/Core.cpp

```diff
@@ -0,0 +1,30 @@
+#include "Core.h"
+
+#include <stdexcept>
+
+std::vector<uint8_t> QToken::possible_values() const {
```

Comment:
Ideally I'd like to rewrite the QToken completely - I don't care if it's HWILDCARD, LWILDCARD or whatever, the only thing that I ever need is the list of possible values. This sounds like a wrapper on `std::vector<uint8_t> ` instead of an enum.

Reply:
:thinking: you say std::vector, I say iterator :smarts:

Reply:
If this was a final solution, sure. But as I've mention, I'd like to get rid of this class completely. The std::vector is absolute non-problem performance-wise, and cost of making a custom iterator hugeley outweights benefits in this case.